api/ruby/enterprise/list_issue_attached_files.rb

```diff
@@ -0,0 +1,30 @@
+require 'octokit'
+
+# Lists all files attached to issues and pull requests on a instance.
+# The list doesn't contain files that are uploaded but not referenced.
+
+## Check for environment variables
+begin
+  access_token = ENV.fetch("GITHUB_TOKEN")
+  hostname = ENV.fetch("GITHUB_HOSTNAME")
+rescue KeyError
+  puts
+  puts "To run this script, please set the following environment variables:"
+  puts "- GITHUB_TOKEN: A valid access token"
+  puts "- GITHUB_HOSTNAME: A valid GitHub Enterprise hostname"
+  exit 1
+end
+
+# Set up Octokit
+Octokit.configure do |kit|
+  kit.api_endpoint = "#{hostname}/api/v3"
+  kit.access_token = access_token
+  kit.auto_paginate = true
+end
+
+pattern = /\[[^\]]*\]\(([^\)]*)\)/
```

Comment:
This is a fairly obtuse regex, a comment indicating what it does (if not how) might be helpful for people looking to learn from this script.